tests/test_bfh.py

```diff
@@ -325,30 +354,96 @@ class Consistent(Mapping):
 
     def test_empty_fields_serialize_as_none_valid_or_no(self):
         """We aren't making assumptions here. Call validate if you want it."""
-        class FirstSchema(Schema):
-            wow = IntegerField(required=True)
-            umm = IntegerField(required=False)
+        first_schema, other_schema, my_map = set_up_schemas()
+        source = first_schema(wow=1)
+        assert source.validate()
 
-        class OtherSchema(Schema):
-            cool = IntegerField(required=True)
-            bad = IntegerField(required=True)  # Uh oh
+        transformed = my_map().apply(source)
+        with self.assertRaises(Invalid):
+            transformed.validate()
 
-        class Mymap(Mapping):
-            source_schema = FirstSchema
-            target_schema = OtherSchema
+        self.assertEqual({"cool": 1}, transformed.serialize())
 
-            cool = Get('wow')
-            bad = Get('umm')
+    def test_empty_fields_serialized_with_defaults(self):
+        first_schema, other_schema, my_map = set_up_schemas(2)
 
-        source = FirstSchema(wow=1)
+        source = first_schema(wow=1, umm=None)
         assert source.validate()
 
-        transformed = Mymap().apply(source)
-        with self.assertRaises(Invalid):
-            transformed.validate()
+        transformed = my_map().apply(source)
+
+        self.assertEqual({"cool": 1, "bad": 2}, transformed.serialize())
+
+#       test that explicit None is different from field not being present.
```

Comment:
Here's maybe where we disagree. I think explicit None should behave exactly the same as `value missing for field`. Note my language. The field is *always* present -- it's a schema. Depending on if you serialize with implicit nulls or explicit nulls, the key containing null value may or may not show up.